packages/core/src/components/ozon-content/nodes/figuur.node.tsx

```diff
@@ -45,6 +45,18 @@ export class OzonContentFiguurNode implements OzonContentNode {
     }
   }
 
+  //Prevent layout shift as an image is downloaded
+  setStyle(breedte?: number, hoogte?: number, schaal?: number) {
+    if (!schaal && breedte && hoogte) {
```

Comment:
Kan je dit uitleggen?

En ik dacht dat de attributen `breedte` en `hoogte` niet toereikend waren voor ons probleem?

Reply:
In het issue heeft @hansgrimm een link naar een Smashing Magazine artikel geplaats: https://github.com/dso-toolkit/dso-toolkit/issues/2701#issuecomment-2180044214. Daarin wordt een oplossing gepresenteerd: zie https://www.smashingmagazine.com/2020/03/setting-height-width-images-important-again/#fixing-the-resizing-problem. 

Kern is:

```css
img {
  width: 100%;
  height: auto;
  aspect-ratio: 16/9;
}
```

Volgens de STOP-standaard zijn `breedte` en `hoogte` verplichte attributen van het `Element tekst:Illustratie`. Met deze 2 attributen kan de `aspect-ratio` berekend worden. En door vervolgens bovenstaande styling in `figuur.node.tsx` op het element `img` te zetten kunnen we de browser de vereist ruimte voor de afbeelding laten reserveren, voordat de `src` van de `img` is geladen.

Echter is het wel zo dat er soms een Illustratie in de xml zit, waarvan breedte en/of hoogte niet juist zijn. In `ozon-response.json` is dat `image2.jpeg`, waarvan de `aspect-ratio` op basis van `breedte` en `hoogte` uit de xml 1.85 is, maar in werkelijkheid 1.37 is. Ook in `ozon-content.content.ts` zit in de content voor "Renvooi-weergave" een Illustratie (klimaatambities_agv_infographic_002_1.jpg), waarvan de `breedte` en de `hoogte` tot een incorrecte aspect-ratio leiden. Dit is ook de reden dat er een diff zit in het ozon-content component op deze Draft-PR:

![Ozon Content -- should render renvooi-weergave elements diff](https://github.com/user-attachments/assets/bd26ce18-f962-49ea-a3e0-145926ae27a9)

Dit probleem denk ik te kunnen oplossen door de css properties `width`, `height` en `aspect-ratio` er weer af te halen op het moment dat de `img` geladen is, zodat dan altijd de natural width en natural height van de img gebruikt gaan worden door de browser.

Wat denk jij @tfrijsewijk?

Reply:
Voordat ik dit in een e2e wil gaan vastleggen wil ik graag van jou @tfrijsewijk feedback over mijn oplossing. Deze heb ik lokaal geverifieerd door naar ... te navigeren, developer tools te openen en op de Network tab throttling op Slow 4G te zetten en vervolgens Empty cache and hard reload te doen en via ctlr+F (cmd+F) naar 6.8 te zoeken. Zo gauw die de DOM is geladen zie je dat de afbeelding onder paragraaf 6.8 Visiekaart langszaam aan het opbouwen/laden is terwijl al wel de benodigde ruimte voor de afbeelding gereserveerd is.

![Screenshot 2024-08-12 at 14 30 07](https://github.com/user-attachments/assets/0ce35578-952f-4333-8b9f-e9d297fe4a71)

Reply:
> Dit probleem denk ik te kunnen oplossen door de css properties width, height en aspect-ratio er weer af te halen op het moment dat de img geladen is, zodat dan altijd de natural width en natural height van de img gebruikt gaan worden door de browser.

Als ik jouw oplossing goed begrijp, zal de afbeelding dan andere formaten aannemen?

Reply:
> > Dit probleem denk ik te kunnen oplossen door de css properties width, height en aspect-ratio er weer af te halen op het moment dat de img geladen is, zodat dan altijd de natural width en natural height van de img gebruikt gaan worden door de browser.
> 
> Als ik jouw oplossing goed begrijp, zal de afbeelding dan andere formaten aannemen?

https://github.com/dso-toolkit/dso-toolkit/pull/2759/files#r1718042710

Reply:
Ok, ik begrijp het:

Ondanks dat we eerder dachten dat de STOP attributen `breedte` en `hoogte` niet toereikend waren (omdat de waardes afweken van de werkelijke afmetingen) ben je erachter gekomen dat in onze testdocumenten de aspect ratio's tussen `breedte`/`hoogte` en de werkelijke breedte en hoogte nagenoeg overeenkomen. Daarom ben je alsnog naar een oplossing met de `img` attributen `width` en `height` gaan kijken.

Ik vind dat een mooie ontdekking en een goede richting. Aanvullend op jouw bevindingen: Als die attributen daadwerkelijk verplicht zijn, zal het mij niet verbazen dat een redacteur (bevoegd gezag) invloed heeft op de waardes en dat we daar "iets" mee moeten doen.

Ik had jouw opmerking 

> Voordat ik dit in een e2e wil gaan vastleggen wil ik graag van jou @tfrijsewijk feedback over mijn oplossing.

Verkeerd geinterpreteerd. Ik dacht dat je nog niets had gedaan en eerst de oplossing wilde overleggen. Ik begreep de functie `onImageLoad()` al niet, ook omdat je verderop zegt:

Ik begrijp wel waarom die functie bestaat: Je gaat in eerste instantie uit van een vaste aspect ratio, en nadat de afbeelding is ingeladen zet je de werkelijke aspect ratio volgens de gerenderde afmetingen.

Maar is jouw volgende opmerking dan een alternatieve oplossing?

> Dit probleem denk ik te kunnen oplossen door de css properties width, height en aspect-ratio er weer af te halen op het moment dat de img geladen is, zodat dan altijd de natural width en natural height van de img gebruikt gaan worden door de browser.

Reply:
Maar maken we dan niet van twee systemen gebruik? In eerste instantie maken we gebruik van STOP `breedte`/`hoogte` met vaste aspect ratio, en daarna van werkelijke afmetingen?

Reply:
> nagenoeg

Zegt maar gerust exact.

Reply:
> een vaste aspect ratio

Ik begrijp niet helemaal wat je met vaste bedoeld. Afbeeldingen hebben 1 aspect-ratio (breedte/hoogte) en in bijna alle gevallen is de aspect-ratio gebaseerd op de stop xml waardes van breedte en hoogte en natural width en height gelijk. En zal dus niet veranderen voor en na het laden van de afbeelding. Alleen heel soms, in het zeldzame geval dat de STOP breedte en hoogte tot een foutieve aspect-ratio leidt, zal dat dan na het laden van de afbeelding gecorrigeerd worden.

Reply:
Voor de volledigheid: Dit hebben we tijdens een meeting besproken.

Ik nam ten onrechte aan dat de vaste ratio `aspect-ratio: 16/9` is overgenomen uit het smashingmagazine artikel. Brainfart.

De overige spraakverwarringen zijn ook besproken. De laatste punten zijn verwerkt als comments.

@iterox dank voor het geduld!